libs/cdk-utils/src/lib/injectors/nodejs-function-defaults-injector.ts

```diff
@@ -143,7 +113,7 @@ export class NodeJsFunctionDefaultsInjector implements IPropertyInjector {
         // If source maps are enabled in our bundling, add the required NODE_OPTIONS flag to the environment
         const environment = {
             ...props.environment,
-            ...(props.bundling?.sourceMap ? { NODE_OPTIONS: '--enable-source-maps' } : {}),
+            ...(this.configuration.sourceMap ? { NODE_OPTIONS: '--enable-source-maps' } : {}),
```

Comment:
Do you think we need to do a proper deep merge with construct props? Technically this would clobber any other NODE_OPTIONS value.

Can be left for future work.